runtime/query.go

```diff
@@ -20,6 +21,29 @@ type Query interface {
 }
 
 func (r *Runtime) Query(ctx context.Context, instanceID string, query Query, priority int) error {
-	// TODO: Add caching here
-	return query.Resolve(ctx, r, instanceID, priority)
+	// take all deps and their last updated time and add them to the query key, prefix query key with instanceID
+	cacheKey := instanceID + query.Key()
+	deps := query.Deps()
+	service, err := r.catalogCache.get(ctx, r, instanceID)
```

Comment:
How will dependant entities be handled here? Is that the responsibility of the query to return them?

Reply:
I mean the use case of a query on the model but parent source has changed where as the model hasn't.

Reply:
Upon reconciliation the last updated time of models derived from a source which has changed will also change. Since we are adding last updated time in the cache key along with the model name, the key will not match with previously cached key and will eventually be evicted from cache. It is the responsibility of the query to list which models its dependent on and then we fetch the last updated time from the catalog for creating the cache key.

Reply:
Ah right we do update the last updated of all child items in reconcile.